go/vt/vtgate/evalengine/expr_compare.go

```diff
@@ -607,18 +612,21 @@ func (l *LikeExpr) eval(env *ExpressionEnv) (eval, error) {
 		return nil, err
 	}
 
+	lbytes, lok := left.(*evalBytes)
+	rbytes, rok := right.(*evalBytes)
+
 	var matched bool
 	switch {
-	case typeIsTextual(left.SQLType()) && typeIsTextual(right.SQLType()):
-		matched = l.matchWildcard(left.(*evalBytes).bytes, right.(*evalBytes).bytes, col.Collation)
-	case typeIsTextual(right.SQLType()):
-		matched = l.matchWildcard(left.ToRawBytes(), right.(*evalBytes).bytes, col.Collation)
-	case typeIsTextual(left.SQLType()):
-		matched = l.matchWildcard(left.(*evalBytes).bytes, right.ToRawBytes(), col.Collation)
+	case lok && rok:
+		matched = l.matchWildcard(lbytes.bytes, rbytes.bytes, col.Collation)
+	case rok:
+		matched = l.matchWildcard(left.ToRawBytes(), rbytes.bytes, col.Collation)
+	case lok:
+		matched = l.matchWildcard(lbytes.bytes, right.ToRawBytes(), col.Collation)
 	default:
 		matched = l.matchWildcard(left.ToRawBytes(), right.ToRawBytes(), collations.CollationBinaryID)
 	}
```

Comment:
Wonder if the above can now be simplified? Since the actual `matchWildcard` call could always use `.ToRawBytes()` (which works on any type), and all we need to compute here is the collation we need to use for the matching?

Reply:
works for me. we were panicking because temporal types are textual but not `*evalBytes`. do you want me to do the change or do you want to just push it on the branch?

Reply:
> works for me. we were panicking because temporal types are textual but not `*evalBytes`. do you want me to do the change or do you want to just push it on the branch?

Yeah, the `*evalTemporal` type was a relatively recent addition we needed to better handle date/time related things. This wasn't then caught I suspect. 

Feel free to push it then!